type-definitions/most.js.flow

```diff
@@ -0,0 +1,410 @@
+
+  constant<B>(b: B): Stream<B>;
+  map<B>(f: (a: A) => B): Stream<B>;
+  tap(f: (a: A) => any): Stream<A>;
+  chain<B>(f: (a: A) => Stream<B>): Stream<B>;
+  flatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
+
+  ap<B, C>(fs: Stream<(a: A) => B>): Stream<C>;
+
+  // Note: Without higher-kinded types, the types for these
+  // cannot be written in a completely safe manner; See https://github.com/Microsoft/TypeScript/issues/1290
+  // For better type safety, consider using most.join/switch/switchLatest with thru
+  join(): A;
+  switch(): A;
+  switchLatest(): A;
+
+  continueWith(f: (a: any) => Stream<A>): Stream<A>;
+  concatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
+  mergeConcurrently<B>(concurrency: number): Stream<B>;
+  merge(...ss: Array<Stream<A>>): Stream<A>;
+  mergeArray(streams: Array<Stream<A>>): Stream<A>;
+
+  combine<B, R>(
+    fn: (a: A, b: B) => R,
+    b: Stream<B>
+  ): Stream<R>;
+  combine<B, C, R>(
+    fn: (a: A, b: B, c: C) => R,
+    b: Stream<B>,
+    c: Stream<C>
+  ): Stream<R>;
+  combine<B, C, D, R>(
+    fn: (a: A, b: B, c: C, d: D) => R,
+    b: Stream<B>,
+    c: Stream<C>,
+    d: Stream<D>
+  ): Stream<R>;
+  combine<B, C, D, E, R>(
+    fn: (a: A, b: B, c: C, d: D, e: E) => R,
+    b: Stream<B>,
+    c: Stream<C>,
+    d: Stream<D>,
+    e: Stream<E>
+  ): Stream<R>;
+
+  combineArray<B, R>(
+    fn: (a: A, b: B) => R,
+    streams: [Stream<B>]
+  ): Stream<R>;
+  combineArray<B, C, R>(
+    fn: (a: A, b: B, c: C) => R,
+    streams: [Stream<B>, Stream<C>]
+  ): Stream<R>;
+  combineArray<B, C, D, R>(
+    fn: (a: A, b: B, c: C, d: D) => R,
+    streams: [Stream<B>, Stream<C>, Stream<D>]
+  ): Stream<R>;
+  combineArray<B, C, D, E, R>(
+    fn: (a: A, b: B, c: C, d: D, e: E) => R,
+    streams: [Stream<B>, Stream<C>, Stream<D>, Stream<E>]
+  ): Stream<R>;
+  combineArray<V, R>(
+    fn: (a: A, ...rest: V[]) => R,
+    streams: Stream<V>[]
+  ): Stream<R>;
+
+  scan<B>(f: (b: B, a: A) => B, b: B): Stream<B>;
+  loop<S, B>(f: (seed: S, a: A) => SeedValue<S, B>, seed: S): Stream<B>;
+
+  concat(s2: Stream<A>): Stream<A>;
+  startWith(a: A): Stream<A>;
+
+  filter(p: (a: A) => boolean): Stream<A>;
+  skipRepeats(): Stream<A>;
+  skipRepeatsWith(eq: (a1: A, a2: A) => boolean): Stream<A>;
+
+  take(n: number): Stream<A>;
+  skip(n: number): Stream<A>;
+  takeWhile(p: (a: A) => boolean): Stream<A>;
+  skipWhile(p: (a: A) => boolean): Stream<A>;
+  skipAfter(p: (a: A) => boolean): Stream<A>;
+  slice(start: number, end: number): Stream<A>;
+
+  until(signal: Stream<any>): Stream<A>;
+  takeUntil(signal: Stream<any>): Stream<A>;
+  since(signal: Stream<any>): Stream<A>;
+  skipUntil(signal: Stream<any>): Stream<A>;
+  during(timeWindow: Stream<Stream<any>>): Stream<A>;
+  throttle(period: number): Stream<A>;
+  debounce(period: number): Stream<A>;
+
+  timestamp(): Stream<TimeValue<A>>;
+  delay(dt: number): Stream<A>;
+
+  await<B>(this: Stream<Promise<B>>): Stream<B>;
```

Comment:
Afaik, using `this` here is valid in TS, but not in flow.  See https://github.com/facebook/flow/issues/452.

Unfortunately, I don't think flow has any way of handling this situation.  So, I think we either have to treat await() and awaitPromises() like join(), or _omit_ the method versions from the type definitions, forcing flow users to call the function versions instead.

Thoughts?